bootloader/bootloader.go

```diff
@@ -226,13 +226,25 @@ type TrustedAssetsBootloader interface {
 	BootChain(runBl Bootloader, kernelPath string) ([]BootFile, error)
 }
 
+// NotScriptableBootloader cannot change the bootloader environment
+// because it supports no scripting or cannot do any writes. This
+// applies to piboot for the moment.
+type NotScriptableBootloader interface {
+	Bootloader
+
+	// Sets boot variables from initramfs - this is needed in
+	// addition to SetBootVars() to prevent side effects like
+	// re-writing the bootloader configuration.
+	SetBootVarsFromInitramfs(values map[string]string) error
+}
+
 // RebootBootloader needs arguments to the reboot syscall when snaps
 // are being updated.
 type RebootBootloader interface {
 	Bootloader
 
 	// GetRebootArguments returns the needed reboot arguments
-	GetRebootArguments() string
+	GetRebootArguments() (string, error)
```

Comment:
https://github.com/snapcore/snapd/pull/11317#discussion_r795775054

:smile:

Reply:
Indeed you were right 🙂